http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/HealthCheckedChannelPool.java

```diff
@@ -159,6 +161,12 @@ private void closeIfUnhealthy(Channel channel) {
      * Determine whether the provided channel is 'healthy' enough to use.
      */
     private boolean isHealthy(Channel channel) {
+        // There might be cases where the channel is not reusable but still active at the moment
+        // See https://github.com/aws/aws-sdk-java-v2/issues/1380
+        if (channel.attr(KEEP_ALIVE).get() != null && !channel.attr(KEEP_ALIVE).get()) {
+            return false;
+        }
+
```

Comment:
I think this is fine, but any reason we don't close the channel after the response ins consumed instead?

Reply:
We do invoke `ChannelHandlerContext#close` in `ResponseHandler` for this case but the channel might not be actually closed and is still active at this point.  It's not guaranteed the channel is closed before it's released into the pool.

Also, because of the fact that the underlying [`HttpStreamsClientHandler`](https://github.com/playframework/netty-reactive-streams/blob/master/netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsClientHandler.java#L66-L73) from netty-reactive-streams does not close the channel if there are still inflight messages, `channel.close` might not be initiated at all. (This is what we discussed in the morning and we might need to figure out why there is still inflight message when the response is complete)